File: CareSync.BusinessAPI/Entities/Billing.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace CareSync.BusinessAPI.Entities;

public class Billing : BaseEntity
{
    public Guid AppointmentId { get; set; }

    public string InvoiceNumber { get; set; } = string.Empty;

    public decimal ConsultationFee { get; set; }

    public decimal MedicineCharges { get; set; }

    public decimal LabCharges { get; set; }

    public decimal OtherCharges { get; set; }

    public decimal TotalAmount { get; set; }

    public decimal PaidAmount { get; set; }

    public decimal DueAmount { get; set; }

    public string PaymentStatus { get; set; } = "Pending";

    public string PaymentMethod { get; set; } = "Cash";

    public DateTime BillDate { get; set; }

    [ForeignKey(nameof(AppointmentId))]
    public Appointment Appointment { get; set; } = null!;
}

File: frontend/src/types/Billing.ts
// src/types/Billing.ts

export type BillingPaymentStatus =
  | 'Paid'
  | 'Pending'
  | 'Cancelled'
  | 'Partially Paid'
  | string;

export interface Billing {
  id: string;

  appointmentId: string;

  invoiceNumber: string;

  patientName: string;

  doctorName: string;

  consultationFee: number;

  medicineCharges: number;

  labCharges: number;

  otherCharges: number;

  totalAmount: number;

  paidAmount: number;

  dueAmount: number;

  paymentStatus: BillingPaymentStatus;

  paymentMethod: string;

  billDate: string;
}

/*
 * ==========================================================
 * CREATE BILLING REQUEST
 * ==========================================================
 */

export interface CreateBillingRequest {
  appointmentId: string;

  consultationFee: number;

  medicineCharges: number;

  labCharges: number;

  otherCharges: number;

  paidAmount: number;

  paymentMethod: string;
}

/*
 * ==========================================================
 * UPDATE BILLING REQUEST
 * ==========================================================
 */

export interface UpdateBillingRequest {
  consultationFee?: number;

  medicineCharges?: number;

  labCharges?: number;

  otherCharges?: number;

  paidAmount?: number;

  paymentMethod?: string;

  paymentStatus?: string;
}

/*
 * ==========================================================
 * API RESPONSES
 * ==========================================================
 */

export interface BillingApiResponse {
  success: boolean;

  message: string;

  data: Billing[];
}

export interface BillingResponse {
  success: boolean;

  message: string;

  data: Billing;
}